src/Hyperion/Benchmark.hs

```diff
@@ -55,7 +60,37 @@ instance Show Benchmark where
   showsPrec x (WithSampling _opt bk) = showsPrec x bk
 
 bench :: String -> Batch () -> Benchmark
-bench name batch = Bench (Text.pack name) batch
+bench name batch = Bench Text.pack (const []) id (const batch) name
+
+benchWith
+  :: (a -> Text)
+  -> (a -> [JSON.Pair])
```

Comment:
That's an option. But let's please start with *concrete* examples. Namely, you mention 3 use cases:

> * Plot the benchmark results of an algorithm (Y axis: time, X axis: input size) to get an idea of the time complexity
> * Linear regression on a known linear-time algorithm
> * Plot the benchmark results of a multi-parameter function against one of the parameters at a time

What would the metadata look like in each case and why would you need to include it? Further, when would a user compute her own linear regression and use the benchmarks DSL to pass that along to the backends?